app/controllers/orders_controller.rb

```diff
@@ -8,35 +8,34 @@ def index
   end
 
   def new
-    order_form = OrderForm.new
-    render 'new', locals: { order_form: order_form, reward: reward, employee: current_employee }
+    @reward = Reward.find(params[:reward])
+    return redirect_to rewards_path, notice: 'You have insufficient funds' if current_employee.points < @reward.price
+
+    @employee = current_employee
+    @address = @employee.address.nil? ? Address.new(employee: @employee) : Address.where(employee: @employee).last
+    @order = Order.new(employee: @employee)
+    render 'new'
   end
 
   def create
-    reward = Reward.find(order_form_params[:reward])
-    if current_employee.points < reward.price
-      redirect_to rewards_path, notice: 'You have insufficient funds'
+    @reward = Reward.find(order_params[:reward_id])
+    @address = Address.new(order_params[:address])
+    @order = Order.new(reward: @reward, reward_snapshot: @reward, address_snapshot: @address, employee: @employee)
+    return redirect_to rewards_path, notice: 'You have insufficient funds' if current_employee.points < @reward.price
+
+    @create_order_service = CreateOrderService.new(order_params)
+    if @create_order_service.call
+      redirect_to orders_path, notice: 'Reward bought'
     else
-      order_form = OrderForm.new(order_form_params)
-      if order_form.save
-        redirect_to orders_path, notice: 'Reward bought'
-      else
-        render 'new',
-               locals: { order_form: order_form, reward: reward, delivery_method: reward.delivery_method,
-                         employee: current_employee }
-      end
+      render 'new', locals: { order: @order, reward: @order.reward, employee: current_employee }
     end
   end
 
   private
 
-  def order_form_params
-    ofp = params.require(:order_form).permit(:reward, :delivery_method, :street, :postcode, :city, :address_id)
-    ofp[:employee] = current_employee
-    ofp
-  end
-
-  def reward
-    Reward.find(params[:reward])
+  def order_params
+    cos = params.require(:order).permit(:reward_id, address: %i[street postcode city])
+    cos[:employee] = current_employee
```

Comment:
check why this is needed?

Reply:
moved employee from params to initialize variable